### PR TITLE
bpo-35054: Add more index entries for symbols.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue35054

### CODE REVIEW

**Review by matrixise — APPROVED** [review]
Hi @serhiy-storchaka 

You can merge your PR, I have just checked on my laptop, the rendering of the index page is fine and the links work.

Good idea.

File: Doc/library/string.rst [hunk 1]
```diff
@@ -349,6 +364,11 @@ meaning in this case.
 The *sign* option is only valid for number types, and can be one of the
 following:
 
+   .. index::
+      single: +; in string formatting
+      single: -; in string formatting
+      single: space; in string formatting
```
Comment (vstinner) [inline Doc/library/string.rst:370]: Sadly "space" goes to https://docs.python.org/dev/genindex-S.html as "space", not as a whitespace. But that's ok ;-)

**Review by vstinner — APPROVED** [review]
LGTM.

"space" goes to the S-Index as "space" (in letters), not as a whitespace. But that's ok. It shouldn't block your PR ;-)

There are a few entries which have the same symbol and same description like os.curdir and os.extsep, but that's ok as well.

File: Doc/reference/simple_stmts.rst [hunk 2]
```diff
@@ -112,6 +112,12 @@ unacceptable.  The rules observed by various types and the exceptions raised are
 given with the definition of the object types (see section :ref:`types`).
 
 .. index:: triple: target; list; assignment
+   single: ,; in target list
+   single: *; in assiggnment target list
```
Comment (pablogsal) [inline Doc/reference/simple_stmts.rst]: s/assiggnment/assignment?

Comment (serhiy-storchaka) [reply Doc/reference/simple_stmts.rst]: Good catch!

**Pull request merged**